applications/StructuralMechanicsApplication/custom_strategies/custom_strategies/eigensolver_strategy.hpp

```diff
@@ -650,6 +660,45 @@ class EigensolverStrategy
         KRATOS_CATCH("")
     }
 
+    /// Normalize the eigenmodes with respect to the mass matrix
+    void MassNormalizeEigenvectors(DenseMatrixType& rEigenvectors)
+    {
+        const SparseMatrixType& rMassMatrix = this->GetMassMatrix();
+        const std::size_t num_modes = rEigenvectors.size1();
+        const std::size_t num_dofs  = rEigenvectors.size2();
+
+        DenseVectorType phi_j(num_dofs);
+        DenseVectorType tmp(num_dofs);
+
+        for (std::size_t j = 0; j < num_modes; ++j) {
+             // copy row j -> phi_j in parallel using IndexPartition
+            IndexPartition<std::size_t>(num_dofs).for_each([&](std::size_t i){
+                phi_j[i] = rEigenvectors(j, i);
+            });
+
+            // tmp = M * phi_j
+            TSparseSpace::Mult(rMassMatrix, phi_j, tmp);
+
+            // Compute modal_mass = phi_j^T * tmp using SumReduction (no atomics)
```

Comment:
```suggestion
            // Compute modal_mass = phi^T * tmp 
```

that comment is even wrong, under the hood it does use atomics

Reply:
Modified